crates/batch-prover/src/proving.rs

```diff
@@ -247,7 +240,7 @@ where
 }
 
 pub(crate) fn state_transition_already_proven<StateRoot, Witness, Da>(
-    input: &BatchProofCircuitInput<StateRoot, Witness, Da::Spec>,
+    input: &BatchProofCircuitInputV2<StateRoot, Witness, Da::Spec>,
```

Comment:
create issue for this function

we don't need to prepare the whole input to understand the same proof has been processed before

Reply:
#1456